components/automate-ui/src/app/pages/project/list/project-list.component.ts

```diff
@@ -87,14 +87,13 @@ export class ProjectListComponent implements OnInit, OnDestroy {
     this.projectsEnabled$ = store.select(atLeastV2p1);
 
     this.projects.applyRulesStatus$
+      .pipe(takeUntil(this.isDestroyed))
       .subscribe(({ state, failed, cancelled, percentageComplete }: ApplyRulesStatus) => {
-        if (state === ApplyRulesStatusState.NotRunning) {
-          if (this.applyRulesInProgress) {
-            this.cancelRulesInProgress = false;
-            this.closeConfirmApplyStopModal();
-          }
-          this.applyRulesInProgress = false;
+        if (this.applyRulesInProgress && state === ApplyRulesStatusState.NotRunning) {
+          this.cancelRulesInProgress = false;
+          this.closeConfirmApplyStopModal();
```

Comment:
This block above is now saying that when the state transitions from running to not-running, close the cancel modal (if it is open).